src/js/components/AppPageComponent.jsx

```diff
@@ -1,9 +1,17 @@
 var _ = require("underscore");
 var classNames = require("classnames");
+var lazy = require("lazy.js");
```

Comment:
This file is at 300+ lines of code if this PR goes through, that's pushing in terms of maintainability. I would like for us to start splitting up files when they start growing to this size.
File size rarely goes down, unless we proactively modularize, split, refactor.